### DISCUSSION TITLE
Vercel limits, serverless functions, and mutations

### DISCUSSION THREAD

**Discussion by aaronshaf** [blitz-js/blitz#1603] [Q&A]
I am looking at the Vercel limit of "12 Serverless Functions Created per Deployment". It seems a sizable Blitz app with, say, a half-dozen tables and corresponding mutations, would surpass this easily.

 Would it make more sense to combine all mutations into one serverless function?

**Comment by Skn0tt** [comment]
Vercel doesn't enforce the 12-Function limit for Next.js-based apps, which Blitz.js is based on. So you shouldn't need to worry about that :)

Besides that, packaging up our code into fewer functions could bring some other improvements (e.g. batched query/mutation calls).
But it would also increase code size and thus cold start times of these functions, which is why we don't do it (yet).

**Reply by flybayer** [reply]
To clarify, Vercel does batch functions together. So the 12 limit still applies, but they consolidate the functions so you are under that.

**Reply by KATT** [reply]
Just stumbled on this. Does the bundling mean one could warm any 1 lambda function [on like a cronjob for instance] make sure all of them are always fast?

**Reply by flybayer** [reply]
@KATT yes
